### PR TITLE
user is re-prompted if they enter a non-integer duration

### LINKED ISSUE #34: If user enters non integer value for reservation days count, program crashes
Program crashes when a user enters a non integer value for the length of their hotel stay

### PR DESCRIPTION
**Pull request by tbuwadi**
added a try catch to re-prompt the user if they enter a non-integer duration... not 100% sure if the logic for this is ok here or if it should be in a model

### CODE REVIEW

File: HotelBooking/Views/book_reservation_view.py [hunk 1]
```diff
@@ -132,8 +132,15 @@ def reserve_room(self):
                     PROMPT_KEY['CUSTOMER'])
             self.start_date = self.prompt_and_get_answer(
                 PROMPT_KEY['START_DATE'])
-            self.duration = int(
-                self.prompt_and_get_answer(PROMPT_KEY['DURATION']))
+            while True:
```
Comment (mclhtay) [inline HotelBooking/Views/book_reservation_view.py]: For validations on simple inputs that do not require controller/model calls, please valid them using PyInquiry's built-in validation parameter:
```
            {
                'type': 'input',
                'message': "Enter transaction amount in cents",
                'name': 'amount',
                'validate': lambda x: x.isdigit() or "Transaction amount invalid",
            }
```

File: HotelBooking/Views/modify_reservation_view.py [hunk 2]
```diff
@@ -147,8 +147,15 @@ def change_duration(self):
                 PROMPT_KEY['LIST_RESERVATION'])
             if answer != "Back":
                 answer_list: List[str] = answer.replace(':', ',').split(',')
-                duration = int(self.prompt_and_get_answer(
-                    PROMPT_KEY['CHANGE_DURATION']))
+                while True:
```
Comment (mclhtay) [inline HotelBooking/Views/modify_reservation_view.py]: ditto.

File: HotelBooking/Views/book_reservation_view.py (hunk 1 above)
Comment (mclhtay) [reply HotelBooking/Views/book_reservation_view.py]: This applies to all other while True validations here.

Comment (tbuwadi) [reply HotelBooking/Views/book_reservation_view.py]: ooo thats so cool

**Comment by tbuwadi** [comment]
@mclhtay changed!

**Review by mclhtay — APPROVED** [review]
Approve on the basis that the author did manual UI testing.

**Comment by mclhtay** [comment]
Also just a reminder, we want to keep the db file in main branch as the default generated version of the sql script. So if this is a modified db file, please remember to run reset-db before merging.

**Comment by tbuwadi** [comment]
> 

hmm okay, i thought the DB was gitignored but maybe not

**Pull request merged**